pkg/quota/image/imagestream_evaluator.go

```diff
@@ -97,41 +106,32 @@ func (c *imageStreamUsageComputer) Usage(object runtime.Object) kapi.ResourceLis
 		return kapi.ResourceList{}
 	}
 
-	images := c.GetImageStreamUsage(is, c.processedImages)
+	specRefs, allRefs := c.GetImageStreamUsage(is, c.processedSpecRefs, c.processedRefs)
 	return kapi.ResourceList{
-		imageapi.ResourceImages: *images,
+		imageapi.ResourceImageStreamTags:   *specRefs,
+		imageapi.ResourceImageStreamImages: *allRefs,
 	}
 }
 
-// makeImageStreamUsageComputerForAdmissionFactory returns an object used during computation of image quota
-// across all repositories in a namespace in a context of admission plugin.
-func makeImageStreamUsageComputerForAdmissionFactory(osClient osclient.Interface) quotautil.UsageComputerFactory {
-	return func() quotautil.UsageComputer {
-		return &imageStreamUsageComputerForAdmission{
-			GenericImageStreamUsageComputer: *NewGenericImageStreamUsageComputer(osClient, true, false),
+// makeImageStreamAdmissionUsageFunc retuns a function for computing a usage of an image stream.
+func makeImageStreamAdmissionUsageFunc(osClient osclient.Interface) generic.UsageFunc {
+	return func(object runtime.Object) kapi.ResourceList {
+		is, ok := object.(*imageapi.ImageStream)
+		if !ok {
+			return kapi.ResourceList{}
 		}
-	}
-}
 
-// imageStreamUsageComputerForAdmission is a context object for use in SharedContextEvaluator
-type imageStreamUsageComputerForAdmission struct {
-	GenericImageStreamUsageComputer
-}
-
-// Usage returns a usage for an image stream.
-func (c *imageStreamUsageComputerForAdmission) Usage(object runtime.Object) kapi.ResourceList {
-	is, ok := object.(*imageapi.ImageStream)
-	if !ok {
-		return kapi.ResourceList{}
-	}
+		c := NewGenericImageStreamUsageComputer(osClient)
 
-	_, imagesIncrement, err := c.GetProjectImagesUsageIncrement(is.Namespace, is, nil)
-	if err != nil {
-		glog.Errorf("Failed to compute project images size increment in namespace %q: %v", is.Namespace, err)
-		return kapi.ResourceList{}
-	}
+		_, specRefsIncrement, _, allRefsIncrement, err := c.GetProjectImagesUsageIncrement(is.Namespace, is, "", false)
```

Comment:
Because this lists all the image streams in the namespace, it will get slower as the number of image streams increases. Do we need to consider a cache of some sort instead? @smarterclayton @derekwaynecarr @deads2k